local-cluster/tests/local_cluster.rs

```diff
@@ -6225,3 +6226,164 @@ fn test_alpenglow_imbalanced_stakes_catchup() {
+    validator_configs[0].turbine_disabled = node_a_turbine_disabled.clone();
+
+    assert_eq!(num_nodes, validator_keys.len());
+
+    // Cluster config
+    let mut cluster_config = ClusterConfig {
+        mint_lamports: total_stake,
+        node_stakes,
+        validator_configs,
+        validator_keys: Some(
+            validator_keys
+                .iter()
+                .cloned()
+                .zip(iter::repeat_with(|| true))
+                .collect(),
+        ),
+        slots_per_epoch,
+        stakers_slot_offset: slots_per_epoch,
+        ticks_per_slot: DEFAULT_TICKS_PER_SLOT,
+        ..ClusterConfig::default()
+    };
+
+    // Create local cluster
+    let cluster = LocalCluster::new_alpenglow(&mut cluster_config, SocketAddrSpace::Unspecified);
+
+    assert_eq!(cluster.validators.len(), num_nodes);
+
+    let vote_pubkeys = validator_keys
+        .iter()
+        .filter_map(|keypair| {
+            cluster
+                .validators
+                .get(&keypair.pubkey())
+                .map(|validator| validator.info.voting_keypair.pubkey())
+        })
+        .collect::<Vec<_>>();
+
+    assert_eq!(vote_pubkeys.len(), num_nodes);
+
+    // Track Node A's votes and when the test can conclude
+    let node_a_filtered_votes = Arc::new(Mutex::new(vec![]));
+    let mut post_experiment_roots = 0;
+
+    // Start vote listener thread to monitor and control the experiment
+    let vote_listener = std::thread::spawn({
+        let mut buf = [0_u8; 65_535];
+        let node_a_filtered_votes = node_a_filtered_votes.clone();
+
+        move || loop {
+            let n_bytes = vote_listener.recv(&mut buf).unwrap();
+            let vote_txn = bincode::deserialize::<Transaction>(&buf[0..n_bytes]).unwrap();
+
+            let (vote_pubkey, parsed_vote, ..) =
+                vote_parser::parse_alpenglow_vote_transaction(&vote_txn).unwrap();
+
+            let node_name = match vote_pubkeys
+                .iter()
+                .position(|pk| pk == &vote_pubkey)
+                .unwrap()
+            {
+                0 => "A",
+                1 => "B",
+                _ => "Unknown",
+            };
+
+            let txn = parsed_vote.as_alpenglow_transaction_ref().unwrap();
+
+            // Once we've received a vote from node B at slot 31, we can start the experiment.
+            if txn.slot() == 31 && node_name == "B" {
+                node_a_turbine_disabled.store(true, Ordering::Relaxed);
+            }
+
+            // Gather all votes issued by node A on slot 32
+            if txn.slot() == 32 && node_name == "A" {
+                node_a_filtered_votes.lock().unwrap().push(*txn);
+            };
+
+            let num_votes = { node_a_filtered_votes.lock().unwrap().len() };
+
+            // We should see a skip followed by a notar fallback. Once we do, the experiment is
+            // complete.
+            if num_votes == 2 {
```

Comment:
Could we have more than 2 votes here? e.g. We are somehow refreshing an earlier vote?